x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonIcon, EuiModal, EuiOverlayMask } from '@elastic/eui';
+import React from 'react';
+import { UpdateTitle, UpdateDescription } from '../properties/helpers';
+import { NOTES_PANEL_WIDTH } from '../properties/notes_size';
+
+import { TimelineTitleAndDescription } from './title_and_description';
+
+export interface SaveTimelineComponentProps {
+  timelineId: string;
+  showOverlay: boolean;
+  toolTip?: string;
+  toggleSaveTimeline: () => void;
+  updateTitle: UpdateTitle;
+  updateDescription: UpdateDescription;
+}
+
+export const SaveTimelineComponent = React.memo<SaveTimelineComponentProps>(
+  ({ timelineId, showOverlay, toggleSaveTimeline, updateTitle, updateDescription }) => (
```

Comment:
can we bind any necessary actions here? I think only `timelineId` should be passed as a prop